sound/soc/sof/ipc4-topology.c

```diff
@@ -354,6 +354,9 @@ static int sof_ipc4_widget_setup_pcm(struct snd_sof_widget *swidget)
 		goto free_available_fmt;
 	}
 
+	if (!WIDGET_IS_AIF(swidget->id))
+		goto skip_gtw_cfg;
+
```

Comment:
I am a bit lost here, we are trying to reuse the same setup_pcm for AIF and internal buffer handling. So do we need a dma_buffer_size here? what does this mean for an internal buffer. Should this test be moved to line 349 instead?

Reply:
you should capture this with a comment, this is useful information that explains the code structure.
this can be done with a fixup